src/Util/ClassSourceManipulator.php

```diff
@@ -0,0 +1,1011 @@
+            $newPropertyBuilder->setDocComment($this->createDocBlock($annotationLines));
+        }
+        $newPropertyNode = $newPropertyBuilder->getNode();
+
+        $this->addNodeAfterProperties($newPropertyNode);
+    }
+
+    private function addGetter(string $propertyName, ?string $returnType, bool $isReturnTypeNullable, array $commentLines = [])
+    {
+        $methodName = 'get'.Str::asCamelCase($propertyName);
+        $getterNodeBuilder = (new Builder\Method($methodName))
+            ->makePublic()
+            ->addStmt(
+                new Node\Stmt\Return_(new Node\Expr\PropertyFetch(new Node\Expr\Variable('this'), $propertyName))
+            )
+        ;
+
+        if (null !== $returnType) {
+            $getterNodeBuilder->setReturnType($isReturnTypeNullable ? new Node\NullableType($returnType) : $returnType);
+        }
+
+        if ($commentLines) {
+            $getterNodeBuilder->setDocComment($this->createDocBlock($commentLines));
+        }
+
+        $this->addMethod($getterNodeBuilder->getNode());
+    }
+
+    private function addSetter(string $propertyName, ?string $type, bool $isNullable, array $commentLines = [])
+    {
+        $builder = $this->createSetterNodeBuilder($propertyName, $type, $isNullable, $commentLines);
+        $this->makeMethodFluent($builder);
+        $this->addMethod($builder->getNode());
+    }
+
+    private function createSetterNodeBuilder(string $propertyName, ?string $type, bool $isNullable, array $commentLines = [])
+    {
+        $methodName = 'set'.Str::asCamelCase($propertyName);
+        $setterNodeBuilder = (new Builder\Method($methodName))->makePublic();
+
+        if ($commentLines) {
+            $setterNodeBuilder->setDocComment($this->createDocBlock($commentLines));
+        }
+
+        $paramBuilder = new Builder\Param($propertyName);
+        if (null !== $type) {
+            $paramBuilder->setTypeHint($isNullable ? new Node\NullableType($type) : $type);
+        }
+        $setterNodeBuilder->addParam($paramBuilder->getNode());
+
+        $setterNodeBuilder->addStmt(
+            new Node\Stmt\Expression(new Node\Expr\Assign(
+                new Node\Expr\PropertyFetch(new Node\Expr\Variable('this'), $propertyName),
+                new Node\Expr\Variable($propertyName)
+            ))
+        );
+
+        return $setterNodeBuilder;
+    }
+
+    /**
+     * @param string $annotationClass The annotation: e.g. "@ORM\Column"
+     * @param array  $options         Key-value pair of options for the annotation
+     *
+     * @return string
+     */
+    private function buildAnnotationLine(string $annotationClass, array $options)
+    {
+        $formattedOptions = array_map(function ($option, $value) {
+            if (is_array($value)) {
+                if (!isset($value[0])) {
+                    // associative array: we'll add this if/when we need it
+                    throw new \Exception('Not currently supported');
```

Comment:
```$value = ["default" => "now()"]``` fires the exception

Reply:
It *is* true that you could use the `ClassSourceManipulator` directly (specifically `addEntityField()`) and pass in some annotations where the value is an associative array. But, this situation is not actually possible in the bundle (at least at this time) - either `addEntityField()` is called by `MakeEntity`, and it passes in only scalar values, or it's called by `EntityRegenerator`, but in that case, the annotations aren't actually printed on the field.

So, it's just not a possible use-case right now :). If someone wants to PR this to fill it in, that's cool. I don't want to fill it in quickly and realize it's incorrect if we *do* in fact later use it.